src/App.js

```diff
@@ -3,29 +3,51 @@ import React, { useState, useEffect } from "react";
 import "./App.css";
 
 function App() {
-    const [messages, setMessages] = useState(["hello", "I love GX!", "where am i????"]);
-    const [newMessage, setNewMessage] = useState("");
+    const [posts, setPosts] = useState([["hello", "I love GX!", "#0fa000","cum"],["hlo", "I hate GX!", "#b7ac59","cum"]]);
+    const [message, setNewMessage] = useState("");
+	const [name, setNewName] = useState("");
+	const [color, setNewColor] = useState("#0fa000");
+	
+	
 
-    const handleNewMsgChange = (e) => {
+    
+	const handleNewMsgChange = (e) => {
         setNewMessage(e.target.value);
     };
+	
+	const handleNameChange = (e) => {
+		setNewName(e.target.value);	
+	}; 
+	
+	const handleColorChange = (e) => {
+		setNewColor(e.target.value);
+	};
+	
     const handleNewMsgSubmit = (e) => {
         e.preventDefault();
-        setMessages([...messages, newMessage]);
+        setPosts([...posts, [message,name,color]]);
```

Comment:
We should use an object for posts rather than an array. Something like `{ message: "hello", name: "DANIEL", color: "#0fa000" }`. That way we don't have to rely on array indexes, and keep track of which index is which value